tests/load/pipeline/test_redshift.py

```diff
@@ -41,3 +43,69 @@ def my_source() -> Any:
         assert pip_ex.value.__cause__.data_type == "time"
     else:
         assert pip_ex.value.__cause__.data_type in ("time", "binary")
+
+AWS_BUCKET = dlt.config.get("tests.bucket_url_s3", str)
+redshift_with_staging_configs = [
+    DestinationTestConfiguration(
+        destination_type="redshift",
+        staging="filesystem",
+        file_format="jsonl",
+        bucket_url=AWS_BUCKET,
+        extra_info="credential-forwarding",
+    ),
+    DestinationTestConfiguration(
+        destination_type="redshift",
+        staging="filesystem",
+        file_format="jsonl",
+        bucket_url=AWS_BUCKET,
+        staging_iam_role="arn:aws:iam::267388281016:role/redshift_s3_read",
+        extra_info="s3-role",
+    ),
+    ## for parquet, region should be unset
+    DestinationTestConfiguration(
+        destination_type="redshift",
+        staging="filesystem",
+        file_format="parquet",
+        bucket_url=AWS_BUCKET,
+        staging_iam_role="arn:aws:iam::267388281016:role/redshift_s3_read",
+        extra_info="s3-role",
+    ),
+]
+@pytest.mark.parametrize( 
+    "destination_config", 
+    redshift_with_staging_configs,
+    ids=lambda x: x.name,
+)
+def test_copy_from_staging_with_region(destination_config: DestinationTestConfiguration ) -> None:
+    """
+    Tests if copy-command is constructed correctly for both iam-role and aws-credentials
+    when REGION is set. It should be set for jsonl, but unset for parquet
+    """
+    # initialize pipeline
+    pipeline : dlt.Pipeline = destination_config.setup_pipeline(
+        "redshift_region_test_" + uniq_id(), dataset_name="redshift_region_test_" + uniq_id()
+    )
+    # TODO assert that region is actually configured
+    # staging_credentials = ## how to get those??
```

Comment:
@sh-rp can you help me get the region from the credentials here? I didnt manage to...